stylelint-rules/no-undeclared-hds-color-tokens.js

```diff
@@ -48,9 +48,11 @@ const plugin = stylelint.createPlugin(
       // https://postcss.org/api/#atrule-walkdecls
       postcssRoot.walkDecls((decl) => {
         // Only look at nodes with a value starting with a token reference
-        if (decl.value.startsWith('var(--token-color')) {
+        const matches = decl.value.match(/var\(--token-color-.*\)/g)
+        const usesHDSColorToken = matches?.length >= 1
+        if (usesHDSColorToken) {
           // Slice out the name of the token
-          const token = decl.value.slice('var('.length, -1)
+          const token = matches[0].slice('var('.length, -1)
```

Comment:
⛏️ we could use a [named capture group](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Ranges) here to make accessing the token name a bit easier.